lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1343,6 +1377,10 @@ using Activity = OptionSet<ActivityFlags>;
 /// indices.
 class DifferentiableActivityInfo {
 private:
+  // TODO: temporary hack while we investigate why reverse mode doesn't work
```

Comment:
Just so it's easier to go back and remove the temporary work around, wherever this comment exists, it should be removed once we figure out how we can use `shouldDifferentiateApplyInst` without special casing for forward or reverse mode. Let me know if there is a better way to mark these locations!

Reply:
Filed [TF-800](https://bugs.swift.org/browse/TF-800) to track activity analysis and "should differentiate" logic differences between forward- and reverse-mode. Edited todo comments to reference TF-800 in https://github.com/apple/swift/pull/26743/commits/9ed94482aa1e62884d3e73f8e3d878e379079f9d.